chore: add generated React hooks and Node client

beeme1mr opened the pull request:
## This PR

- adds generated React hooks based on the flag manifest
- adds generated Node.js client based on the flag manifest
- adds generate NPM script
- adds generate Make command
- extends `flag.json` to include flag manifest requirements

beeme1mr committed:
feat: add codegen react

Signed-off-by: Michael Beemer <beeme1mr@users.noreply.github.com>

beeme1mr committed:
add generate flags npm script

Signed-off-by: Michael Beemer <beeme1mr@users.noreply.github.com>

lukas-reining reviewed (commented):
This is really cool!

beeme1mr commented:
Unfortunately, SSG fails when using suspense flags. I believe the root cause is the no op provider is never in a ready state (by design). This causes the build process to hang indefinitely.

beeme1mr committed:
Merge branch 'main' into code-gen

Signed-off-by: Michael Beemer <beeme1mr@users.noreply.github.com>

beeme1mr committed:
chore: update package dependencies and refactor feature flag integration

- Updated package.json to use new versions of OpenFeature SDKs and other dependencies.
- Refactored API routes to utilize the generated OpenFeature client instead of the deprecated server SDK methods.
- Changed imports in checkout and header components to use the new generated hooks for feature flags.
- Created new generated client files for both Node.js and React to streamline feature flag usage.
- Updated database connection logic to align with the new feature flag client methods.

Signed-off-by: Michael Beemer <beeme1mr@users.noreply.github.com>
